sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -2237,13 +2238,18 @@ private void copy(
             + "retries: %d", jobIdPrefix, Bound.MAX_RETRY_JOBS));
       }
 
-      private void removeTemporaryTables(DatasetService tableService,
-          List<TableReference> tempTables) throws Exception {
+      static void removeTemporaryTables(DatasetService tableService,
+          List<TableReference> tempTables) {
         for (TableReference tableRef : tempTables) {
-          tableService.deleteTable(
-              tableRef.getProjectId(),
-              tableRef.getDatasetId(),
-              tableRef.getTableId());
+          try {
+            LOG.debug("Deleting table " + toJsonString(tableRef));
```

Comment:
in logger syntax, no need to concat strings. Instead:

```java
LOG.debug("Deleting table {}", toJsonString(tableRef));
```

This will not run the concat in the event that `debug` level is off.

(same below)